YAKL_mem_transfers.h

```diff
@@ -16,7 +16,7 @@
       hipMemcpyAsync(dst,src,elems*sizeof(T),hipMemcpyDeviceToHost,0);
       check_last_error();
     #elif defined (YAKL_ARCH_SYCL)
-      sycl_default_stream.memcpy(dst, src, elems*sizeof(T));
+      sycl_default_stream.memcpy(dst, src, elems*sizeof(T)).wait();
```

Comment:
These were meant to be async. Should I keep the `.wait()` ?

Reply:
removed `.wait()`